### ISSUE TITLE
Automated Docker builds don't appear to be triggering

### ISSUE THREAD

**Issue by jcormier** [sameersbn/docker-redmine#474]
@sameersbn  Could you check to see if something needs attention?

Tag 4.1.4 and 4.2.2 don't appear to have triggered a build.

**Comment by sameersbn** [comment]
Hey @jcormier . Turns out docker has disabled automated builds for free accounts.
I work on updating the CI to build and push the image over the weekend.

**Comment by frank-dspeed** [comment]
@sameersbn migration to github tasks is the solution. and push it from there to docker and github registry.

**Comment by frank-dspeed** [comment]
@sameersbn we need 2 values configured as secrets so that i can enable pushing to the docker registry when ever something hits master so everything should be tested before it hits master

The login-action options required for Docker Hub are:

username and password: This is your Docker Hub username and password. We recommend storing your Docker Hub username and password as secrets so they aren't exposed in your workflow file. For more information, see "Creating and using encrypted secrets." https://docs.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets

it would be best if you would create them with that names
```
         username: ${{ secrets.DOCKER_USERNAME }}
          password: ${{ secrets.DOCKER_PASSWORD }}
```

**Comment by frank-dspeed** [comment]
https://github.com/sameersbn/docker-redmine/blob/frank-dspeed-github-actions-build/.github/workflows/docker-image.yml

**Comment by jcormier** [comment]
> i can enable pushing to the docker registry when ever something hits master

Can the github workflow trigger on pushed tags like how the old docker builds worked?  This allows building tagged docker images and support updates to older redmine versions.

It also triggered on pushes to master which got labeled as :latest

**Comment by frank-dspeed** [comment]
@jcormier at present i did configure it to trigger build push on github release creation that is maybe the most easy flow and matches the old one?

**Comment by jcormier** [comment]
Add okay that should work fine.  Thanks.

**Comment by jcormier** [comment]
@sameersbn Looks like you build and pushed 4.1.4 and 4.2.2 today?  Are these automated?  Any info on how you've set it up if so?

**Comment by sameersbn** [comment]
Hey. Since I was unable to find the time to work on the automation I manually built and pushed the images.

**Comment by jcormier** [comment]
#488 Request to push 4.2.3-1

**Comment by sameersbn** [comment]
working in the releases...

**Comment by sameersbn** [comment]
✅ sameersbn/redmine:4.2.3-1
✅ sameersbn/redmine:4.2.3-2

**Comment by sameersbn** [comment]
https://github.com/sameersbn/docker-redmine/pull/491 implements gh actions for building docker images when commits occur on the master branch and when new tags are created. Additionally the tags are published to docker.io, quay.io and ghcr.io (which will show up under `packages` link of the repo which is pretty neat).

Sorry about the delay in getting this done.

**Comment by jcormier** [comment]
This is great, thanks

**Comment by jcormier** [comment]
#494

**Comment by jcormier** [comment]
Tried delete pushing the 4.1.6 tag and repushing.  Build didn't trigger.  Then created a new tag 4.1.6-1 and pushed tag but build didn't trigger.

```
git push --delete origin 4.1.6
git push origin 4.1.6
```

**Comment by jcormier** [comment]
Tried switching to triggering on releases but the release for 4.1.6-1 didn't trigger a github action...

**Comment by jcormier** [comment]
@sameersbn Could you manually trigger build for 4.1.6-1 and 4.2.4

**Comment by frank-dspeed** [comment]
@jcormier can you define the final behavior that you want and i simply integrate it simply only tell me the commands that should lead to a release and i integrate it.

**Comment by frank-dspeed** [comment]
@jcormier 4.1.6-1 is not semver compatible! semver allows only chars after the - not digits!4.1.6-pre0 would be valid but there must be a char after -

**Comment by sameersbn** [comment]
Sorry about the delay. I'm taking a look. From my initial tests the builds
should have triggered

**Comment by sameersbn** [comment]
@frank-dspeed is right. the tags need to be semver compatible. so I propose using `-rX` for the revisions, fe. `4.1.6-r1`

**Comment by sameersbn** [comment]
there were a couple of issues:
1. as @frank-dspeed mentioned the tags need to be semver compatible.
2. the ci configuration did not exist in the `4.1-stable` branch. as a result tags created from that branch were not being picked up by the CI

I have fixed the above issues and created the tag `4.1.6-r2` on the 4.1-stable branch. for `4.2.4` release, I deleted the existing tag and recreated it from the tip of the master

**Comment by sameersbn** [comment]
i can see the tags are now built and also available on the github releases page https://github.com/sameersbn/docker-redmine/pkgs/container/redmine. 

Again.. sorry about the delay.

**Comment by frank-dspeed** [comment]
@jcormier invited you as admin  jcormier has been added as a collaborator with admin permissions on the repository. https://github.com/dockerimages/docker-redmine

i will configure now automated publishing and will integrate the rest of my stashed changes directly we will produce the best redmine image ever and will automate the maintainance. we will publish it https://hub.docker.com/u/dockerimages/redmine

**Comment by jcormier** [comment]
> @jcormier invited you as admin jcormier has been added as a collaborator with admin permissions on the repository. https://github.com/dockerimages/docker-redmine
@sameersbn is a busy guy so I see the value in having a shared repo where more of us have admin rights to keep things working. Though getting people to move over will be a bit bumpy.
> 
> i will configure now automated publishing and will integrate the rest of my stashed changes directly we will produce the best redmine image ever and will automate the maintainance. we will publish it https://hub.docker.com/u/dockerimages/redmine
Any large structural changes I'd like a chance to look over before it makes it into master. I don't have a large amount of time to redesign the project or anything.

**Comment by sameersbn** [comment]
P.s. i will check if the sameersbn project can be converted to a org. So
possibly it should handle the multiple admins requirements

**Comment by frank-dspeed** [comment]
@sameersbn you can have my "dockerimages" org if you like i had never the time and contributors to maintain all that.

**Comment by frank-dspeed** [comment]
Closed by #498
